packages/internals/src/views/handleViewsIO.ts

```diff
@@ -0,0 +1,117 @@
+import * as E from 'fp-ts/lib/Either'
+import { pipe } from 'fp-ts/lib/function'
+import * as T from 'fp-ts/lib/Task'
+import * as TE from 'fp-ts/lib/TaskEither'
+import path from 'path'
+import { match } from 'ts-pattern'
+
+import {
+  createDirIfNotExists,
+  getFilesInDir,
+  getFoldersInDir,
+  removeDir,
+  removeFile,
+  writeFile,
+} from '../utils/fs-functional'
+
+export interface IntrospectionViewDefinition {
+  // The database or schema where the view is located
+  schema: string
+
+  // The name of the view
+  name: string
+
+  // The database SQL query that defines the view
+  definition: string
+}
+
+type HandleViewsIOParams = {
+  views: IntrospectionViewDefinition[]
+  schemaPath: string
+}
+
+/**
+ * For any given view definitions, the CLI must either create or update the corresponding view definition files
+ * in the file system, in `${path.dirname(schemaPath)}/views/{viewDbSchema}/{viewName}.sql`.
+ * If some other files or folders exist within the `views` directory, the CLI must remove them.
```

Comment:
Why delete other files that are not conflicting with what we want to create?

Reply:
Yeah, this feels a little strange to me too

Reply:
(This would be lines 79 to 89)

Reply:
We do not know what views used to exist in the database during introspection, so we return back a list of views what are there now. The file is deleted from the folder if it's not part of the views what we return.

Reply:
Agree with Julius. Closing this as "solved" as it's working like documented in Notion